### PR TITLE
Service registry glossary entry

### LINKED ISSUE #1783: Need a glossary entry for Service Registry
We assume readers understand "service registry" in a few places. A crisp definition with expected behavior would be useful.

### PR DESCRIPTION
**Pull request by frankbu**
Fixes #1783

### CODE REVIEW

File: content/help/glossary/service-registry.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: Service Registry
+---
+
+Istio maintains an internal service registry containing the set of [services](#service),
+and their corresponding [service endpoints](#service-endpoint), running in a service mesh.
+Most services are automatically added to the registry by platform-specific Pilot plug-ins,
```
Comment (rshriram) [inline content/help/glossary/service-registry.md]: Pilot adapters (plug-ins have a different meaning for Pilot). Might also want to specify that Istio is merely reading the service registry to generate Envoy configuration. Its not providing service discovery (service discovery is the responsibility of the underlying platform (k8s/consul/plain DNS).

File: content/help/glossary/service-registry.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+but additional services can be registered manually using a
+[ServiceEntry](/docs/concepts/traffic-management/#service-entries) configuration.
+
+The service registry manifests itself as the set of available clusters that Pilot
+provides to [Envoy](#envoy), via the CDS API, which it uses to forward service requests.
```
Comment (rshriram) [inline content/help/glossary/service-registry.md]: Do we need to talk about this internal implementation detail?

File: content/help/glossary/tls-origination.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ title: TLS Origination
 ---
 
 TLS origination occurs when an Istio proxy (sidecar or egress gateway) is configured to accept unencrypted
-internal HTTP connections, and then encrypt the request before forwarding it to either a simple or mutual TLS
-secured HTTPS server. This is the opposite of [TLS termination](https://en.wikipedia.org/wiki/TLS_termination_proxy)
+internal HTTP connections, encrypt the requests, and then forward them to either simple or mutual TLS
```
Comment (rshriram) [inline content/help/glossary/tls-origination.md]: suggest rephrasing /breaking up the sentence.. 
and then forward them to a HTTPS server thats secured using simple or mutual TLS.

File: content/help/glossary/service-registry.md (hunk 2 above)
Comment (frankbu) [reply content/help/glossary/service-registry.md]: Yes, I was wondering about that myself.

@geeknoid, can you provide some context? What docs do you think might want to reference this glossary entry?

**Pull request merged**